### DISCUSSION TITLE
Vector-based plots possible in HTML output?

### DISCUSSION THREAD

**Discussion by ramiromagno** [quarto-dev/quarto-cli#1481] [Q&A]
When exporting to HTML format, I expected the R code chunk option `#| fig-format: svg` to make my plots SVG based instead of PNG but it does not seem to be happening. My intention is to have vector based plots in a website --- it would look nicer.

**Comment by mcanouil** [comment]
It seems `fig-format: svg` does not work as a cell option.

- This will produce a SVG figure.

  ````
  ---
  title: "Title"
  format: html
  fig-format: svg
  ---
  
  ## Quarto
  
  ```{r}
  library(ggplot2)
  ggplot()
  ```
  ````

- This won't (likely a bug):

  ````
  ---
  title: "Title"
  format: html
  ---
  
  ## Quarto
  
  ```{r}
  #| fig-format: svg
  library(ggplot2)
  ggplot()
  ```
  ````

**Reply by ramiromagno** [reply]
Thanks for the quick reply! Indeed using `fig-format` as a global option in the YAML header works (that works for me, for now).

**Reply by mcanouil** [reply]
I'll let the Quarto's dev determine if it's intended or indeed a bug (otherwise it's definitely a feature request of mine to have access to this option locally).

**Reply by alonsoC1s** [reply]
I'm using Julia as the language and tried using `fig-format: pdf`, but just can't seem to get high quality figures with Julia. No issue with any other language

**Comment by jjallaire** [comment]
It's technically possible for us to set that per cell for Knitr however NOT for Jupyter (which is probably why we didn't implement it for Knitr). Note that you can use the `dev: svg` cell option with Knitr to do this now.

**Reply by mcanouil** [reply]
How does `fig-format` work with `dev`?
Does it set the device for all engine thus it overwrites `dev` globally when knitr engine is used?
(Mostly curious)

**Comment by jjallaire** [comment]
It sets the default knitr chunk option (which can then be overridden per
chunk)
